src/main/java/org/opensearch/knn/index/engine/MethodComponent.java

```diff
@@ -328,19 +330,42 @@ public static Map<String, Object> getParameterMapWithDefaultsAdded(
         Map<String, Object> parametersWithDefaultsMap = new HashMap<>();
         Map<String, Object> userProvidedParametersMap = methodComponentContext.getParameters();
         Version indexCreationVersion = knnMethodConfigContext.getVersionCreated();
+        Mode mode = knnMethodConfigContext.getMode();
+        CompressionLevel compressionLevel = knnMethodConfigContext.getCompressionLevel();
+
+        boolean isOnDiskWithBinaryQuantization = mode == Mode.ON_DISK
```

Comment:
why we are checking for on_disk? we should just look for compression parameter to set these parameters.

Reply:
I was trying to safe ...and going was definition...It does not harm ..

Reply:
but then it is not consistent. The idea is BQ is lossy so we want to increase the hyper parameter values.

Reply:
Addressed!!